recipes/cgns/all/conanfile.py

```diff
@@ -1,92 +1,125 @@
 import os
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
+from conan.errors import ConanException, ConanInvalidConfiguration
+from conan.tools.files import get, copy, rmdir, apply_conandata_patches
 
 
-required_conan_version = ">=1.29.1"
-
+required_conan_version = ">=1.43.0"
 
 class CgnsConan(ConanFile):
     name = "cgns"
     description = "Standard for data associated with the numerical solution " \
                   "of fluid dynamics equations."
-    topics = ("conan", "cgns", "data", "cfd", "fluids")
+    topics = ("cgns", "data", "cfd", "fluids")
     homepage = "http://cgns.org/"
     license = "Zlib"
     url = "https://github.com/conan-io/conan-center-index"
     settings = "os", "compiler", "build_type", "arch"
-    exports_sources = ["CMakeLists.txt", "patches/**"]
-    generators = "cmake"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
-        "with_hdf5": [True, False]
+        "with_hdf5": [True, False],
+        "parallel": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
-        "with_hdf5": True
+        "with_hdf5": True,
+        "parallel": False,
     }
 
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", src=".", dst=self.export_sources_folder)
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, patch["patch_file"], src=".", dst=self.export_sources_folder)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def validate(self):
+        if self.info.options.parallel and not (self.info.options.with_hdf5 and self.info.options["hdf5"].parallel):
+            raise ConanInvalidConfiguration("The option 'parallel' requires HDF5 with parallel=True")
+        if self.info.options.parallel and self.info.options.with_hdf5 and self.info.options["hdf5"].enable_cxx:
+            raise ConanInvalidConfiguration("The option 'parallel' requires HDF5 with enable_cxx=False")
+
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
-        del self.settings.compiler.libcxx
+        try:
+            # In windows, with msvc, the compiler.libcxx doesn't exist, so it will raise.
```

Comment:
which exception does it raise? I'm curious because Visual Studio does not have libcxx too and so far, we don't have such problem. Maybe, it's a case to be compatible.

Reply:
I think its a 2.0 change where it now fails when the settings are missing (I don't recall where I read that though)